vlasovsolver/cpu_trans_map_amr.cpp

```diff
@@ -987,78 +1168,104 @@ void getSeedIds(const dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGr
+            // is thus the velocity space boundary
+            for (uint i=0; i<WID3; ++i) {
+               blockValues[i] = block_data[cellid_transpose[i]];
+            }
+
+            // now load values into the actual values table..
+            uint offset =0;
+            for (uint k=0; k<WID; k++) {
+               for(uint planeVector = 0; planeVector < VEC_PER_PLANE; planeVector++){
+                  // store data, when reading data from data we swap dimensions 
+                  // using precomputed plane_index_to_id and cell_indices_to_id
+                  values[i_trans_ps_blockv_pencil(planeVector, k, b, lengthOfPencil)].load(blockValues + offset);
+                  offset += VECL;
+               }
+            }
+         } else {
+            for (uint k=0; k<WID; ++k) {
+               for(uint planeVector = 0; planeVector < VEC_PER_PLANE; planeVector++) {
+                  values[i_trans_ps_blockv_pencil(planeVector, k, b, lengthOfPencil)] = Vec(0);
+               }
             }
          }
       } else {
-         for (uint k=0; k<WID; ++k) {
-            for(uint planeVector = 0; planeVector < VEC_PER_PLANE; planeVector++) {
-               values[i_trans_ps_blockv_pencil(planeVector, k, b, lengthOfPencil)] = Vec(0);
+         // Source cell can be at higher refinement level than pencil (P::amrTransSplitPencilsOnlyForFace==true)
+         Realf blockValues[WID3] = { 0.0 };
+         // Average source information over 4 sibling cells. Find siblings:
+         auto mySiblings = mpiGrid.get_all_children(mpiGrid.get_parent(source_neighbors[b + VLASOV_STENCIL_WIDTH]));
+         auto myIndices = mpiGrid.mapping.get_indices(source_neighbors[b + VLASOV_STENCIL_WIDTH]);
+         uint siblingCount=0;
+         for (uint i_sib = 0; i_sib < MAX_NEIGHBORS_PER_DIM; ++i_sib) {
+            auto sibling = mySiblings.at(i_sib);
+            auto sibIndices = mpiGrid.mapping.get_indices(sibling);
+
+            // Only use siblings at same position in pencil
+            if (myIndices.at(dimension) == sibIndices.at(dimension)) {
+               // Get cell pointer and local block id
+               SpatialCell* srcCell = mpiGrid[sibling];
+               const vmesh::LocalID blockLID = srcCell->get_velocity_block_local_id(blockGID,popID);
+               if (blockLID != srcCell->invalid_local_id()) {
+                  const Realf* block_data = srcCell->get_data(blockLID,popID);
+                  for (uint i=0; i<WID3; ++i) {
+                     blockValues[i] += block_data[cellid_transpose[i]];
+                  }
+                  siblingCount++;
+               }
+            }
+         }
+         if (siblingCount>0) {
+            // Rescale with sibling count (should be 4)
+            if (siblingCount>1) {
+               const Realf rescale = 1./siblingCount;
+               for (uint i=0; i<WID3; ++i) {
+                  blockValues[i] *= rescale;
+               }
+            }
```

Comment:
This can be optimised away, either put it in the loop before or in the loop after to avoid one full-block-loop, no? And this sibling count is anyway 4, no?

Reply:
Emphasis on the word "should be". But indeed we could put a 0.25 rescaling factor hardcoded in there and just thrown an error if siblingcount is not equal to 4.